torch/csrc/autograd/functions/accumulate_grad.h

```diff
@@ -100,11 +100,13 @@ struct TORCH_API AccumulateGrad : public Node {
       if (!GradMode::is_enabled() &&
           !new_grad.is_sparse() &&
           new_grad.use_count() <= num_expected_refs &&
-          utils::obeys_layout_contract(new_grad, variable)) {
+          ((!new_grad.is_mkldnn() && utils::obeys_layout_contract(new_grad, variable))
```

Comment:
HOw can grad be an mkldnn tensor, when mkldnn_convolution_backward_weights always converts grad to dense tensor? https://github.com/pytorch/pytorch/blob/b7449f9a21cc799b7f5632db26b74991991668df/aten/src/ATen/native/mkldnn/Conv.cpp#L193

Reply:
hi, @ngimel, Could you give me more details about it? I can't catch your meaning.

Reply:
This function (`accumulateGrad`) gets grads returned backward functions (in case of mkl convolutions that would be `mkl_convolution_backward_weights`). Since `mkl_convolution_backward_weights` always returns dense tensors, `new_grad` cannot be mkldnn tensor, unless I'm missing something?

Reply:
@ngimel, yes, **mkldnn_convolution_backward_weight**s always returns dense tensors, but for **mkldnn_convolution_backward_input**,  it returns mkldnn tensor.